sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ResolveWriteToStream.scala

```diff
@@ -139,8 +139,8 @@ object ResolveWriteToStream extends Rule[LogicalPlan] with SQLConfHelper {
               .stripMargin)
         }
       }
-      val checkpointDir = checkpointPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
-      fs.mkdirs(checkpointDir)
+      val checkpointDir = fileManager.getQualifiedCheckpointPath()
+      fileManager.mkdirs(checkpointDir)
```

Comment:
For this particular case, the API we need is `createCheckpointDirectory`. But I feel it's also good to expose the qualified checkpoint path with an API.

Reply:
You mean adding both `createCheckpointDirectory` and `getQualifiedCheckpointPath`? If we just use `createCheckpointDirectory` here, `getQualifiedCheckpointPath` is not used at all.

Reply:
How about we add `createCheckpointDirectory` first? We can add `getQualifiedCheckpointPath` later when needed.

Reply:
sgtm